File: data/src/main/java/org/cryptomator/data/cloud/s3/S3CloudApiExceptions.java
package org.cryptomator.data.cloud.s3;

public class S3CloudApiExceptions {

	public static boolean isAccessProblem(String errorCode) {
		return errorCode.equals(S3CloudApiErrorCodes.ACCESS_DENIED.getValue()) || errorCode.equals(S3CloudApiErrorCodes.ACCOUNT_PROBLEM.getValue()) || errorCode.equals(S3CloudApiErrorCodes.INVALID_ACCESS_KEY_ID.getValue());
	}

	public static boolean isNoSuchBucketException(String errorCode) {
		return errorCode.equals(S3CloudApiErrorCodes.NO_SUCH_BUCKET.getValue());
	}
}

File: data/src/main/java/org/cryptomator/data/cloud/s3/S3CloudApiExceptions.kt
package org.cryptomator.data.cloud.s3

object S3CloudApiExceptions {

	@JvmStatic
	fun isAccessProblem(errorCode: String): Boolean {
		return errorCode == S3CloudApiErrorCodes.ACCESS_DENIED.value //
				|| errorCode == S3CloudApiErrorCodes.ACCOUNT_PROBLEM.value //
				|| errorCode == S3CloudApiErrorCodes.INVALID_ACCESS_KEY_ID.value
	}

	@JvmStatic
	fun isNoSuchBucketException(errorCode: String): Boolean {
		return errorCode == S3CloudApiErrorCodes.NO_SUCH_BUCKET.value
	}
}
